### DISCUSSION TITLE
Disable subfields during different events based on user role

### DISCUSSION THREAD

**Discussion by shuvoworld** [Laravel-Backpack/community-forum#834] [Q&A (Help)] [answered]
How can I disable subfields or one subfield/repeat fields based on different user role?

**Comment by jcastroa87** [comment]
Hello @shuvoworld 

Maybe you can try to use [CrudField JavaScript Library](https://backpackforlaravel.com/docs/6.x/crud-fields-javascript-api#1-show-hide-a-field) to hide or show fields.

Cheers.

**Comment by o15a3d4l11s2** [comment, marked as answer]
It really depends on the use-case.
- If you want to dynamically disable them while the form is open, then as @jcastroa87 suggested - you can leverage [CrudField JavaScript Library](https://backpackforlaravel.com/docs/6.x/crud-fields-javascript-api#1-show-hide-a-field).
- If you just need to have the fields statically disabled/enabled based on the logged in user role, then you can simply use the `disabled` attribute for the corresponding fields when defining the fields in `setupCreateOperation()`, `setupUpdateOperation()` accordingly following the [field attributes](https://backpackforlaravel.com/docs/6.x/crud-fields#optional-field-attributes-for-presentation-purposes-1) guide. As an example
  ```php
    protected function setupUpdateOperation()
    {
        $attributes = [];

        if (!backpack_user()->isSuperAdmin) {
            $attributes = ['disabled' => 'disabled'];
        }

        // Repeatable from the examples
        CRUD::field([
            'name'  => 'testimonials',
            'label' => 'Testimonials',
            'type'  => 'repeatable',
            'subfields' => [
                [
                    'name'    => 'name',
                    'type'    => 'text',
                    'label'   => 'Name',
                    'wrapper' => ['class' => 'form-group col-md-4'],
                ],
                [
                    'name'    => 'position',
                    'type'    => 'text',
                    'label'   => 'Position',
                    'wrapper' => ['class' => 'form-group col-md-4'],
                    'attributes' => $attributes, // the subfield will conditionally be disabled based on the user
                ],
            ],
        ]);
    }
   ```
    But in case you want only a specific repeatable item (i.e. only 3-rd repeatable row) to be disabled, this example will not help you, as it defines the enabled/disabled field for all repeatable rows.

**Reply by shuvoworld** [reply]
Thank you a lot!
